src/test/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtilsUnitTest.java

```diff
@@ -1733,30 +1737,197 @@ public void testTotalPloidy(final int[] ploidies, final int defaultPloidy, final
+                {".bcf", ".idx", false, true},
+                {".bcf", ".idx", true, false},
+
+                {".vcf.bgz", ".tbi", true, true},
+                // AbstractFeatureReader fails to recognize this as block compressed unless it has an accompanying index
+                // Is that correct behavior ?
+                //".vcf.bgz", ".tbi", false, true},
+                {".vcf.gz", ".tbi", false, true},
+                {".vcf.bgz", ".tbi", true, false},
+
+                // defaults to .vcf
+                {".tmp", ".idx", false, true},
+        };
+    }
+    @Test(dataProvider = "createVCFWriterData")
+    public void testCreateVCFWriterWithOptions(
+            final String outputExtension,
+            final String indexExtension,
+            final boolean createIndex,
+            final boolean createMD5) throws IOException {
+
+        final File tmpDir = createTempDir("createVCFTest");
+        final File outputFile = new File(tmpDir.getAbsolutePath(), "createVCFTest" + outputExtension);
+
+        Options options[] = createIndex ?
+                new Options[] {Options.INDEX_ON_THE_FLY} :
+                new Options[] {};
+        try (final VariantContextWriter writer = GATKVariantContextUtils.createVCFWriter(
+                outputFile,
+                makeSimpleSequenceDictionary(),
+                createMD5,
+                options)) {
+            writeHeaderAndBadVariant(writer, false);
+        }
+
+        final File outFileIndex = new File(outputFile.getAbsolutePath() + indexExtension);
+        final File outFileMD5 = new File(outputFile.getAbsolutePath() + ".md5");
+
+        Assert.assertTrue(outputFile.exists());
+        Assert.assertEquals(outFileIndex.exists(), createIndex);
+        Assert.assertEquals(outFileMD5.exists(), createMD5);
+
+        verifyFileType(outputFile, outputExtension, createIndex);
+    }
+
+    // just make sure we can read the file with the corresponding codec
+    private void verifyFileType(final File resultVCFFile, final String outputExtension, final boolean createIndex) {
+        AbstractFeatureReader<?, ?> afr = null;
+        if (outputExtension.equals(".vcf")) {
+            afr= AbstractFeatureReader.getFeatureReader(
+                    resultVCFFile.getAbsolutePath(),
+                    new VCFCodec(),
+                    createIndex);
+        }
+        else if (outputExtension.equals(".bcf")) {
+            afr = AbstractFeatureReader.getFeatureReader(
+                    resultVCFFile.getAbsolutePath(),
+                    new BCF2Codec(),
+                    createIndex);
+        }
+        else if (outputExtension.equals(".vcf.bgz") ||
+                outputExtension.equals(".vcf.gz") ||
+                outputExtension.equals(".tmp")) {
+            afr = AbstractFeatureReader.getFeatureReader(
+                    resultVCFFile.getAbsolutePath(),
+                    new VCFCodec(),
+                    createIndex);
+        }
```

Comment:
Would be simpler to just construct a `FeatureDataSource` and make sure that you don't get an exception, instead of building the reader manually like this.